### ISSUE TITLE
decode-md5 is failing on mingw32 / Visual Studio 2010 (32bits)

### ISSUE THREAD

**Issue by gcode-importer** [uclouvain/openjpeg#292]
Originally reported on Google Code with ID 292

```
As seen here:


http://my.cdash.org/testDetails.php?test=14151717&build=581070

CMake Error at /home/voxxl/Dashboards/MyTests/openjpeg-experimental/tests/nonregression/checkmd5refs.cmake:64
(message):
  not equal: [8afcac9a696cc8d753b0eb9f4ae692ff
  CT_Phillips_JPEG2K_Decompr_Problem.j2k_0.pgx] vs
  [91bcc4158e0c6af59950b6290d1abdea
  CT_Phillips_JPEG2K_Decompr_Problem.j2k_0.pgx]
```

Reported by [malaterre](https://github.com/malaterre) on 2014-03-14 08:51:46

**Comment by mayeut** [comment]
Same kind of errors with kakadu update: http://my.cdash.org/viewTest.php?onlydelta&buildid=800503
those *-compare_dec-ref-out2base are failing now that I updated kakadu

**Comment by mayeut** [comment]
Strange thing is that travis OS X build doesn't fail with the same version of kakadu...
http://my.cdash.org/buildSummary.php?buildid=823322
Maybe linked to OS version

**Comment by detonin** [comment]
@malaterre @mayeut Is there anything to be done at openjpeg level to prevent these computation errors ?

**Comment by malaterre** [comment]
this may be a long debate, but at a minimum I would start with -ffloat-store and/or -std=c99 just to check this is not a silly mistake. x86_64 use SSE by default, while -m32 uses 387... I've seen report that ` -ffast-math` is supported now, this issue may have gone away.

**Comment by detonin** [comment]
Thanks. What shall be done to reproduce the issue ? Do we need a mingw32 build ?

**Comment by malaterre** [comment]
no, simply a `-m32` and `-m64` should be enough (if linux/maxosx is your preferred OS)

**Comment by detonin** [comment]
I confirm the issue is still valid, on Windows (see IPX1 machine on CDash), MacOSX and Debian. The delta between Debian 64 bits and Debian 32 bits (with -m32 flag) is available here:
http://my.cdash.org/viewTest.php?onlydelta&buildid=900134

**Comment by ValZapod** [comment]
Is this reproducable on Visual Studio 2022? LOL.
